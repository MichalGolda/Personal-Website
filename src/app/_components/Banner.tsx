import { SecondaryButton } from "@/app/_components";

export default function Banner() {
  return (
    <div className="bg-primary w-full py-16 px-8 flex flex-col gap-y-8 text-center items-center">
      <h2 className="text-4xl font-bold text-white max-w-3xl">
        Masz pomysł na projekt? Skontaktuj się ze mną i stwórzmy razem coś
        wyjątkowego
      </h2>
      <div className="flex flex-col gap-y-1">
        <h4 className="text-2xl text-white font-semibold">
          <span className="text-[#B6C2FF80] font-bold text-5xl px-2">”</span>
          Początek jest najważniejszą częścią pracy
        </h4>
        <span className="text-white text-base text-secondary font-medium">
          Platon
        </span>
      </div>
      <a href="#contact">
        <SecondaryButton className="w-fit" type="button">
          Skontaktuj się
        </SecondaryButton>
      </a>
    </div>
  );
}
