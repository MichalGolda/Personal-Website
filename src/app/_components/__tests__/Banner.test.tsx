import "@/app/_mocks/intersectionObserver.mock";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Banner from "../Banner";

describe("Banner component", () => {
  it("renders without erros", () => {
    render(<Banner />);
  });

  it("renders unchanged", () => {
    const { container } = render(<Banner />);

    expect(container).toMatchSnapshot();
  });
});
