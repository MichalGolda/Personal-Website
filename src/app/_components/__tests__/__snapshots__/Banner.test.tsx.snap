// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders unchanged 1`] = `
<div>
  <div
    class="bg-primary w-full py-16 px-4 flex flex-col gap-y-8 text-center items-center"
  >
    <h2
      class="text-2xl md:text-3xl lg:text-4xl font-bold text-white max-w-3xl"
    >
      Masz pomysł na projekt? Skontaktuj się ze mną i stwórzmy razem coś wyjątkowego
    </h2>
    <div
      class="flex flex-col gap-y-1"
    >
      <h4
        class="text-lg md:text-xl lg:text-2xl text-white font-semibold"
      >
        <span
          class="text-[#B6C2FF80] font-bold text-3xl md:text-4xl lg:text-5xl px-2"
        >
          ”
        </span>
        Początek jest najważniejszą częścią pracy
      </h4>
      <span
        class="text-white text-sm md:text-base text-secondary font-medium"
      >
        Platon
      </span>
    </div>
    <a
      href="#contact"
    >
      <button
        class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-secondary text-white hover:bg-secondary/90 active:bg-secondary/80 border border-transparent w-fit"
        type="button"
      >
        Skontaktuj się
      </button>
    </a>
  </div>
</div>
`;
